Fixes machines becoming invisible when breaking or replacing multiblock parts

Saereth opened the pull request:
### What

The Root Cause was MachineRenderState relying on identity-based lookups (IdentityHashMap in model rendering, IdMapper in network sync).

However, StateHolder.setValue() only returns interned instances when called on an already-interned state. When non-interned  states entered the system via deserialization or network sync, identity lookups failed, causing blocks to render nothing.

## Implementation Details

- MachineRenderState - Added intern() method that looks up the canonical state instance from StateDefinition.getPossibleStates(), and wired it into the CODEC via xmap() so deserialized states are always interned
  - MetaMachineBlockEntity - setRenderState() now validates the state belongs to the correct definition and interns it; onLoad() also validates since @persisted bypasses the setter.
  - MachineRenderStatePayload - Interns state before registry ID lookup during network write
  - MultiblockControllerMachine - onPartUnload() now immediately invalidates the structure when a part is removed, ensuring all remaining parts have their render state updated to unformed
  - MachineModel - Added fallback rendering that searches by property values when identity lookup fails, preventing invisible blocks if a non-interned state slips through and is mostly a failsafe

## Outcome

- Multiblock renders are now consistent across reloads and break/replace operations with no more invisible blocks due to invalid states.

Saereth committed:
Fixes multiblock rendering issues on world load

Fixes #4276
Addresses an issue where multiblocks sometimes fail to render correctly after initial chunk loading due to the block entity not being fully initialized when the chunk is first rendered.

Forces a model data refresh on the client side when a block entity finishes loading to ensure correct rendering of multiblocks, especially in cases where the chunk was rendered before the block entity was fully available. Also, it avoids returning null model render types, which could cause the block to be invisible, by instead returning the render types of the default model.

Saereth committed:
Fixes machines becoming invisible when breaking or replacing multiblock parts

The Root Cause was MachineRenderState relying on identity-based lookups (IdentityHashMap in model rendering, IdMapper in network sync).

However, StateHolder.setValue() only returns interned instances when called on an already-interned state. When non-interned  states entered the system via deserialization or network sync, identity lookups failed, causing blocks to render nothing.

The Fix:
- MachineRenderState - Added intern() method that looks up the canonical state instance from StateDefinition.getPossibleStates(), and wired it into the CODEC via xmap() so deserialized states are always interned
  - MetaMachineBlockEntity - setRenderState() now validates the state belongs to the correct definition and interns it; onLoad() also validates since @Persisted bypasses the setter.
  - MachineRenderStatePayload - Interns state before registry ID lookup during network write
  - MultiblockControllerMachine - onPartUnload() now immediately invalidates the structure when a part is removed, ensuring all remaining parts have their render state updated to unformed
  - MachineModel - Added fallback rendering that searches by property values when identity lookup fails, preventing invisible blocks if a non-interned state slips through and is mostly a failsafe

Saereth committed:
Merge branch 'GregTechCEu:re/1.21-with-love' into re/1.21-with-love

Saereth committed:
forgot to clean this up when I was debugging

Saereth committed:
Move MachineRenderState to api/machine package

Fixes dedicated server crash caused by client-side class loading. MachineRenderState was in client/model/machine but used by server-side code (MetaMachineBlockEntity, MachineRenderStatePayload, etc.). When the dedicated server loaded these classes, it triggered  loading of other client-only classes in that package, causing LDLRegisterClient and GuiGraphics class loading errors.

Reabstraction commented on src/main/java/com/gregtechceu/gtceu/api/machine/MachineRenderState.java:
```diff
@@ -11,10 +9,31 @@
+                return interned;
+            }
+        }
+        return this; // Fallback if not found (shouldn't happen)
```
Consider adding a one time debug log whenever this fallback path is taken

Saereth replied:
added

Saereth committed:
Fix multiblock parts not showing formed textures on world load

-  Added @Persisted to controllerPositions in MultiblockPartMachine so controller references are available immediately on load
- Add onLoad() to rebuild controllers from persisted positions and
  clear IS_FORMED if controllers can't be resolved (prevents inconsistent visual state with overlays but no texture overrides)
- Fix MetaMachineBlockEntity.onLoad() to always intern() the render state after NBT load, ensuring it matches modelsByState keys

Saereth committed:
Retrigger CI

Saereth committed:
Retrigger CI

jurrejelle committed:
Merge branch 're/1.21-with-love' into re/1.21-with-love

Saereth committed:
added onetime fallback debug logging

Saereth committed:
Merge remote-tracking branch 'origin/re/1.21-with-love' into re/1.21-with-love

jurrejelle committed:
[DOCS] [1.21.1] Use mike for multi-version support (#4314)

Reabstraction committed:
Merge remote-tracking branch 'origin/1.21' into re/1.21-with-love
